agent/api/container.go

```diff
@@ -216,6 +221,45 @@ func (c *Container) GetKnownExitCode() *int {
 	return c.knownExitCode
 }
 
+// SetRegistryAuthCredentials will set the credentials for pulling image from ecr
+func (c *Container) SetRegistryAuthCredentials(credential *credentials.IAMRoleCredentials) error {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	if c.RegistryAuthentication == nil || c.RegistryAuthentication.ECRAuthData == nil {
+		return fmt.Errorf("Container: set ecr pull credentials failed, no registry auth data for container")
```

Comment:
Can you reformat the error messages as per https://github.com/golang/go/wiki/CodeReviewComments#error-strings?